pkg/ccl/changefeedccl/changefeed_processors.go

```diff
@@ -1234,7 +1234,7 @@ func (cf *changeFrontier) Start(ctx context.Context) {
 	cf.js = newJobState(nil, cf.flowCtx.Cfg.Settings, cf.metrics, timeutil.DefaultTimeSource{})
 
 	if cf.spec.JobID != 0 {
-		job, err := cf.flowCtx.Cfg.JobRegistry.LoadClaimedJob(ctx, cf.spec.JobID)
+		job, err := cf.flowCtx.Cfg.JobRegistry.LoadJob(ctx, cf.spec.JobID)
```

Comment:
Not updating the the job from a machine that doesn't have the lease has, in the past at least, been a nice layer of protection against confusion caused by two copies of the job updating it concurrently.

I don't think we should ship this with this in place.

Rather, we might pursue one of the following:

1) Send the current job's lease in the spec so that we can continue to check that lease here. It might take some updates to the job code to make that work well but they should be minor.

2) Send the status updates we want to do back to the coordinator and do them there.

I worry that (2) might defeat some of the point of this change and be a rather large change to how this job works, so I prefer (1).

Reply:
yeah, I think I prefer (1) as well, since writing to the job record might be "work" and I would like all "work" to be able to be scheduled off the coordinator. 

I suppose (2) could do the frontier work and just send simple-to-apply writes back to the coordinator to commit for it, but that just seems complicated.